src/Router.sol

```diff
@@ -514,24 +475,40 @@ contract Router is ReentrancyGuard, IRouter, Initializable, UUPSUpgradeable, Acc
 
     /// @notice Sets the minimum contract upgrade delay
     /// @param _minimumContractUpgradeDelay The new minimum delay for upgrade operations
-    function setMinimumContractUpgradeDelay(uint256 _minimumContractUpgradeDelay) external onlyAdmin {
-        require(_minimumContractUpgradeDelay > 2 days, ErrorsLib.UpgradeDelayTooShort());
-        minimumContractUpgradeDelay = _minimumContractUpgradeDelay;
-        emit MinimumContractUpgradeDelayUpdated(minimumContractUpgradeDelay);
+    function setMinimumContractUpgradeDelay(uint256 _minimumContractUpgradeDelay)
+        public
+        override (IRouter, ScheduledUpgradeable)
+        onlyAdmin
+    {
+        super.setMinimumContractUpgradeDelay(_minimumContractUpgradeDelay);
     }
 
     /// @notice Updates the swap request BLS signature validator contract
     /// @param _swapRequestBlsValidator The new swap request BLS validator contract address
-    function setSwapRequestBlsValidator(address _swapRequestBlsValidator) external onlyAdmin {
+    /// @param signature The BLS signature authorising the update
+    function setSwapRequestBlsValidator(address _swapRequestBlsValidator, bytes calldata signature) external {
+        require(_swapRequestBlsValidator != address(0), ErrorsLib.ZeroAddress());
+        uint256 nonce = ++currentNonce;
```

Comment:
This is a bit counterintuitive because the Ethereum convention is that you sign the current nonce and then increment it., But this is also a matter of opinion. (This would also require changing the client and tests)
```suggestion
        uint256 nonce = currentNonce++;
```

Reply:
I used `uint256 nonce = ++currentNonce;` to avoid using zero as nonce. Just as personal preference to kind of differentiate between “no nonce” and “first nonce”. But I'm happy to change it, if others prefer.
@CluEleSsUK @azixus

Reply:
don't have a strong opinion - all options seem fine